File: backend/src/main/java/com/mini/credit/dto/caisse/SessionCaisseAnomalieResponse.java
package com.mini.credit.dto.caisse;

import com.mini.credit.enums.StatutDossierAnomalieSession;
import com.mini.credit.enums.StatutSessionCaisse;
import com.mini.credit.enums.TypeAnomalieSessionCaisse;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Builder
public class SessionCaisseAnomalieResponse {
    private Long id;
    private Long sessionId;
    private TypeAnomalieSessionCaisse typeAnomalie;
    private StatutSessionCaisse ancienStatut;
    private StatutSessionCaisse nouveauStatut;
    private String motif;
    private String commentaire;
    private StatutDossierAnomalieSession statutDossier;
    private String demandePar;
    private String validePar;
    private LocalDateTime dateDemande;
    private LocalDateTime dateValidation;
    private String actionExecutee;
}

File: frontend/src/app/features/caisse/models/session-caisse-anomalie-response.ts
import { StatutSessionCaisse } from './statut-session-caisse';

export type TypeAnomalieSessionCaisse =
  | 'OUVERTURE_ERRONEE_SANS_MOUVEMENT'
  | 'PRE_CLOTURE_ERRONEE'
  | 'CLOTURE_ERRONEE_SANS_MOUVEMENT'
  | 'SESSION_AVEC_MOUVEMENTS_RECTIFICATION_REQUISE';

export type StatutDossierAnomalieSession = 'DEMANDEE' | 'VALIDEE' | 'REJETEE' | 'EXECUTEE';

export interface SessionCaisseAnomalieResponse {
  id: number;
  sessionId: number;
  typeAnomalie: TypeAnomalieSessionCaisse;
  ancienStatut: StatutSessionCaisse;
  nouveauStatut?: StatutSessionCaisse;
  motif: string;
  commentaire?: string;
  statutDossier: StatutDossierAnomalieSession;
  demandePar?: string;
  validePar?: string;
  dateDemande: string;
  dateValidation?: string;
  actionExecutee?: string;
}
